src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,199 @@
+            );
+        }
+        Task toDo = new Todo(stripDescription(input)[0]);
+        Duke.list.add(toDo);
+        System.out.println("Got it. I've added this task:");
+        System.out.println(toDo.toString());
+        System.out.println("Now you have " + Duke.list.size() + " in the list.");
+    }
+
+    /**
+     * Adds a task of Event type into the list.
+     * 
+     * 
+     * @param input
+     * @throws DukeException
+     */
+    protected static void event(String input) throws DukeException {
+        if (stripDescription(input)[0] == "") {
+            throw new DukeException(
+                    "Oops, the description of event cannot be empty!"
+            );
+        }
+
+        if (stripDescription(input)[1] == "") {
+            throw new DukeException(
+                    "Oops, please tell me where this will occur!"
+            );
+        }
+
+        LocalDate date = LocalDate.parse(stripDescription(input)[1]);
+        Task event = new Event(stripDescription(input)[0], date);
+        System.out.println("Got it. I've added this task:");
+        Duke.list.add(event);
+        System.out.println(event.toString());
+        System.out.println("Now you have " + Duke.list.size() + " in the list.");
+    }
+
+    protected static void list() {
+
+    }
+
+    /**
+     * Deletes a task from the list.
+     * 
+     * @param input
+     * @throws DukeException
+     */
+    protected static void delete(String input) throws DukeException {
+        ArrayList<String> arr = new ArrayList<>(Arrays.asList(input.split(" ")));
+        if (arr.get(1) == "") {
+            throw new DukeException(
+                    "Oops! Which task would you like to delete?"
+            );
+        }
+
+        int i = Integer.parseInt(arr.get(1));
+        if (i <= Duke.list.size() && i > 0) {
+            System.out.println("Noted. I've removed this task:");
+            System.out.println(Duke.list.get(i-1).toString());
+            Duke.list.remove(i - 1);
+            System.out.println("Now you have " + Duke.list.size() + " tasks in the list.");
+
+        } else {
+            throw new DukeException(
+                    "That is not a valid task number!"
+            );
+        }
+    }
+
+    /**
+     * To process the input string and separate the description and date (if applicable).
+     * @param s
+     * @return a string array [description, date] where date="" if not applicable.
+     * @throws DukeException
+     */
+    private static String[] stripDescription(String s) throws DukeException {
+
+        ArrayList<String> arr = new ArrayList<>(Arrays.asList(s.split(" ")));
+        if (arr.contains("/at")) {
+            int i = arr.indexOf("/at");
+            String str1 = String.join(" ", arr.subList(1, i));
+            String str2 = String.join(" ", arr.subList(i + 1, arr.size()));
+            String[] str = new String[2];
+
+
+
+            str[0] = str1;
+            str[1] = str2;
```

Comment:
maybe you can write a separate private method to pass in the parameters to get your result ? As i notice you had applied the same code for other options of this method. It can make the overall code less repetitive.